ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -1061,9 +1125,24 @@ void handleBody(InputStream payload, MessageMetadata messageMetadata, MessageInf
         blobType = blobData.getBlobType();
         chunkIndexToBuffer = new TreeMap<>();
         if (blobType == BlobType.MetadataBlob) {
-          handleMetadataBlob(blobData, userMetadata, encryptionKey);
+          if (rawMode) {
+            // Send the list of chunk Ids
+            chunkIndexToBuffer.put(0, blobData.getStream().getByteBuffer());
```

Comment:
That's a good question.  I had it that way originally and changed it because something was going wrong.  Let me try it again with rawPayloadBuffer and see what happens since I don't recall.

Reply:
Changed to rawPayloadBuffer and updated the logic that was causing tests to fail.